ldj-readme-renderer/render.js

```diff
@@ -8,45 +8,57 @@
 const fs = require('node:fs');
 const process = require('node:process');
 
-const OUTPUT = `README.md`;
+const OUTPUT = `../README.md`;
 
-const TEMPLATE = `[![Coverage](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=coverage)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-[![Technical Debt](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=sqale_index)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-[![Reliability Rating](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=reliability_rating)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-[![Maintainability Rating](https://sonarcloud.io/api/project_badges/measure?project=Ali-chakaroun_ISEP-LivingDocumentation&metric=sqale_rating)](https://sonarcloud.io/summary/new_code?id=Ali-chakaroun_ISEP-LivingDocumentation)
-# Living Documentation Java
-
-This project is an extension of the [Living Documentation][ldoc] set of tools to
-Java. As part of the Living Documentation toolchain, it can be used to generate
-documentation and diagrams from source code so that documentation is always
-up-to-date.
-
-## Analyzer
-
-The analyzer parses Java projects into Living Documentation JSON files that can
-be rendered by Living Documentation renderers. It accepts the following command
-line options:
+const json = JSON.parse(fs.readFileSync(process.argv[2]));
 
+// Section for rendering the command line options
+const CMD_LINE_TABLE_TEMPLATE = `
 | Short option | Long option | Description |
 | ------------ | ----------- | ----------- |
 ~
-
-[ldoc]: https://github.com/eNeRGy164/LivingDocumentation
 `;
 
-function render(d) {
+function renderCmdOptions(d) {
   if (d.Name === 'addOption' || d.Name === 'addRequiredOption') {
     const args = d.Arguments.map(a => a.Text).map(JSON.parse);
     const short = args[0] ? `\`-${args[0]}\`` : '';
     const long = args[1] ? `\`--${args[1]}\`` : '';
     const required = d.Name.includes('Required') ? '(Required.)' : '';
     return `| ${short} | ${long} | ${args[3]} ${required} |\n`;
   } else if (d instanceof Array) {
-    return d.map(render).join('');
+    return d.map(renderCmdOptions).join('');
   } else if (d instanceof Object) {
-    return Object.values(d).map(render).join('');
+    return Object.values(d).map(renderCmdOptions).join('');
   }
 }
 
-const json = JSON.parse(fs.readFileSync(process.argv[2]));
-fs.writeFileSync(OUTPUT, TEMPLATE.replace('~', () => render(json)));
+const cmdOptionsTable = CMD_LINE_TABLE_TEMPLATE.replace('~', () => renderCmdOptions(json));
+
+// Section rendering the implemented visited methods
+
+const JAVAPARSER_DOC_SITE = 'https://javadoc.io/static/com.github.javaparser/javaparser-core/3.25.6/';
```

Comment:
Probably not a big difference, but we use version 3.25.1, not 3.25.6 (see analyzer-java's pom.xml).